### PR TITLE
uplink_olt_model_add

### PR DESCRIPTION
**Pull request by AAm-kun**
## Description

Added UPLINK olt support. V-SOL olt also might work as uplink is rebrand of v-sol in some way. not tested on V-SOL though.

Some part of the code was taken from https://gist.github.com/mdpuma/0df4bc4fe6dd8d0aad6d0445902a9f63 and some was generated by Mistral AI.

P:S: Not a ruby dev, just a network engineer who was able to make it work and wanted to share with the community.

### CODE REVIEW

File: lib/oxidized/model/uplinkolt.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      
+    cmd 'configure terminal' do
+      # Enter configure terminal mode
+      cmd 'show version' do |cfg|
+      cfg.gsub! /^show version/, ''
+      comment cfg
+      end
+      
+    end
+    cmd 'show time' do |cfg|
```
Comment (systeembeheerder) [inline lib/oxidized/model/uplinkolt.rb:28]: why? This way your config changes every time oxidized pulls?

Comment (AAm-kun) [reply lib/oxidized/model/uplinkolt.rb:28]: well it was not in my initial code. Then i saw it somewhere while going through other models, so putted it. 
Was having the same confusion that was it necessary or not. Now as you have pointed out I am removing it.

**Comment by robertcheramy** [comment]
Please document your change in CHANGELOG.md and add the model in docs/[Supported-OS-Types.md](https://github.com/ytti/oxidized/blob/master/docs/Supported-OS-Types.md)

The code must pass `rubocop` and `rake test`.
Have a look at: https://github.com/ytti/oxidized/blob/master/CONTRIBUTING.md#code-like-a-ruby-professional

**Comment by AAm-kun** [comment]
> Please document your change in CHANGELOG.md and add the model in docs/[Supported-OS-Types.md](https://github.com/ytti/oxidized/blob/master/docs/Supported-OS-Types.md)
> 
> The code must pass `rubocop` and `rake test`. Have a look at: https://github.com/ytti/oxidized/blob/master/CONTRIBUTING.md#code-like-a-ruby-professional

Finally made time to do the test and updated the files today. Let me know if everything is fine or not. 
Thank you.

**Comment by systeembeheerder** [comment]
@AAm-kun you should ask @robertcheramy to review, not me. 


I find i hard to find any information on the vendor 'uplink' and 'EP4440' seems a specific device, not an network OS. Is OLT the OS then? But yet you add EP4440 to the OS list. Just my two cents.

**Comment by AAm-kun** [comment]
@systeembeheerder  Well **Uplink** is a rebrand of **VSOL** if I am not wrong. The OS is same, at least what I saw the webui in youtube videos.
Was not sure what to put in the OS, as there are nothing about this device on google. But saw in the supported list there were some os same name as devices that was the reasoning for putting **EP4440**. **EP4440** is not a specific device either, it is a prefix of multiple devices which shares the same OS. This is basically what my device shows.
image

@robertcheramy please review when you have time.

**Comment by robertcheramy** [comment]
The PR looks good, I will be happy to merge it into master after following changes:
- Please provide a [unit test](https://github.com/ytti/oxidized/blob/master/docs/Creating-Models.md#create-unit-tests-for-the-model) for the model. You will need to rebase / merge from master in order to use `model_helper.rb` in the unit test.
- Please document your changes in CHANGELOG.md

**Comment by robertcheramy** [comment]
Update - consider the unit test point as optional - making the yaml file is to complicated yet, I must improve the documentation before asking submitter do to so.
Please just document your changes in CHANGELOG.md

**Comment by AAm-kun** [comment]
> Update - consider the unit test point as optional - making the yaml file is to complicated yet, I must improve the documentation before asking submitter do to so. Please just document your changes in CHANGELOG.md

change-log has been added. Let me know if anything else is missing.

**Comment by robertcheramy** [comment]
Sorry for the long delay - I was busy with other topics.

**Pull request merged**